python/hail/representation/genotype.py

```diff
@@ -63,7 +68,7 @@ def __repr__(self):
             (self.gt, self.ad, self.dp, self.gq, self.pl)
 
     def __eq__(self, other):
-        return self._jrep.equals(other._jrep)
+        return isinstance(other, Genotype) and self._jrep.equals(other._jrep)
```

Comment:
I don't think so. Did you have a use case in mind?

Reply:
I was thinking that if we don't want to allow this it might be better to just have
a `@typecheck` to make sure that `other` is of type `nullable(Genotype)` so an error gets thrown?

Reply:
I don't think that's the behavior we want. I think we'd want the return value to be False and not have a type error raised.

Reply:
we'll raise an error in Expression comparisons, the way we do in the string expr language now. But having == throw an error for normal python comparisons might be rude